dialogue-core/src/main/java/com/palantir/dialogue/core/RetryingChannel.java

```diff
@@ -40,29 +40,34 @@
 
     private final Channel delegate;
     private final int maxRetries;
+    private final boolean propagateQoS;
 
-    RetryingChannel(Channel delegate, int maxRetries) {
+    RetryingChannel(Channel delegate, int maxRetries, boolean propagateQoS) {
         this.delegate = delegate;
         this.maxRetries = maxRetries;
+        this.propagateQoS = propagateQoS;
     }
 
     @Override
     public ListenableFuture<Response> execute(Endpoint endpoint, Request request) {
-        return new RetryingCallback(delegate, endpoint, request, maxRetries).execute();
+        return new RetryingCallback(delegate, endpoint, request, maxRetries, propagateQoS).execute();
     }
 
     private static final class RetryingCallback {
         private final Channel delegate;
         private final Endpoint endpoint;
         private final Request request;
         private final int maxRetries;
+        private final boolean propagateQoS;
         private int failures = 0;
 
-        private RetryingCallback(Channel delegate, Endpoint endpoint, Request request, int maxRetries) {
+        private RetryingCallback(
+                Channel delegate, Endpoint endpoint, Request request, int maxRetries, boolean disableQosRetries) {
```

Comment:
could we actually just plumb this enum into the RetryingChannel?

Reply:
I went back and forth on this. Feels a little odd to pipe the ClientConfiguration bits this deep, but an enum is cleaner. I'm okay with reusing the enum since RetryingChannel isn't public.